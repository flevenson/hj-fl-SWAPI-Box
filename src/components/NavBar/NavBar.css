button {
  background: none;
  border: none;
  font-size: 2rem;
  cursor: pointer;
  font-family: 'Fugaz One', cursive;
  color: #4A3934;
}

.selected {
  color: #FEFFC0;
}

.not-selected {
  color: #4A3934;
}