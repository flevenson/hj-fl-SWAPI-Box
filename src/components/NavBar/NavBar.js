import React from 'react';
import PropTypes from 'prop-types';
import './NavBar.css'

const NavBar = ({ selected, getData }) => {

	const handleClick = (event) => {
		let buttonName = event.target.getAttribute('name')
		getData(buttonName)
	}
  
  return (
    <div className="nav-bar-div">
      <button className={selected === 'people' ? 'selected' : 'people'} name='people' onClick={handleClick}>People</button>
      <button className={selected === 'planets' ? 'selected' : 'planets'} name='planets' onClick={handleClick}>Planets</button>
      <button className={selected === 'vehicles' ? 'selected' : 'vehicles'} name='vehicles' onClick={handleClick}>Vehicles</button>
      <button className={selected} name='favorites'>Favorites</button>
    </div>
  )
  
}

NavBar.propTypes = {
	selected: PropTypes.string,
	getPeople: PropTypes.func

}

export default NavBar