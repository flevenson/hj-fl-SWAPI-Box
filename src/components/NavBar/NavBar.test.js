import React from 'react';
import { shallow, mount } from 'enzyme';

import NavBar from './NavBar.js';

describe('NavBar', () => {
	let wrapper;
	let mockFunction;

	beforeEach(() => {
	  mockFunction = jest.fn()
	  wrapper = shallow(<NavBar 
	  	handleNavClick={mockFunction} 
	  	selected={''} 
	  />);		
	})

	it('should match snapshot', () => {
	  expect(wrapper).toMatchSnapshot();
	})

	it('should call handleClick when people is clicked', () => {
	  wrapper.find('.people').simulate('click')
	  expect(mockFunction).toBeCalled()
	})

	it('should call handleClick when planets is clicked', () => {
	  wrapper.find('.planets').simulate('click')
	  expect(mockFunction).toBeCalled()
	})

	it('should call handleClick when vehicles is clicked', () => {
	  wrapper.find('.vehicles').simulate('click')
	  expect(mockFunction).toBeCalled()
	})

	// it('should only allow one button to be selected', () => {
	//   wrapper = mount(<NavBar handleNavClick={mockFunction} selected={''} /> );
	//   let selected = wrapper.find('.selected')
	//   wrapper.find('.people').simulate('click')
	//   selected = wrapper.find('.selected')
	//   console.log(selected)
	// })	
})