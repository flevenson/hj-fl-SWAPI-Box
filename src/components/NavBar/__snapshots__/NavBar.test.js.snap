// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar
    getData={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="people"
          name="people"
          onClick={[Function]}
        >
          People
        </button>,
        <button
          className="planets"
          name="planets"
          onClick={[Function]}
        >
          Planets
        </button>,
        <button
          className="vehicles"
          name="vehicles"
          onClick={[Function]}
        >
          Vehicles
        </button>,
        <button
          name="favorites"
        >
          Favorites
        </button>,
      ],
      "className": "nav-bar-div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "People",
          "className": "people",
          "name": "people",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "People",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Planets",
          "className": "planets",
          "name": "planets",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Planets",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Vehicles",
          "className": "vehicles",
          "name": "vehicles",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Vehicles",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Favorites",
          "className": undefined,
          "name": "favorites",
        },
        "ref": null,
        "rendered": "Favorites",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="people"
            name="people"
            onClick={[Function]}
          >
            People
          </button>,
          <button
            className="planets"
            name="planets"
            onClick={[Function]}
          >
            Planets
          </button>,
          <button
            className="vehicles"
            name="vehicles"
            onClick={[Function]}
          >
            Vehicles
          </button>,
          <button
            name="favorites"
          >
            Favorites
          </button>,
        ],
        "className": "nav-bar-div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "People",
            "className": "people",
            "name": "people",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "People",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planets",
            "className": "planets",
            "name": "planets",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Planets",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicles",
            "className": "vehicles",
            "name": "vehicles",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorites",
            "className": undefined,
            "name": "favorites",
          },
          "ref": null,
          "rendered": "Favorites",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
